src/test/javascript/spec/component/file-upload-submission/file-upload-submission.spec.ts

```diff
@@ -0,0 +1,171 @@
+        expect(comp.acceptedFileExtensions.replace(/\./g, '')).to.be.equal(fileUploadExercise.filePattern);
+        expect(comp.fileUploadExercise).to.be.equal(fileUploadExercise);
+        expect(comp.isAfterAssessmentDueDate).to.be.true;
+        expect(comp.numberOfAllowedComplaints).to.be.undefined;
+        expect(comp.submission).to.be.undefined;
+
+        const maxScore = debugElement.query(By.css('div p strong'));
+        expect(maxScore).to.exist;
+        expect(maxScore.nativeElement.textContent).to.be.equal(`Max. Score: ${fileUploadExercise.maxScore}`);
+
+        // check if fileUploadInput is available
+        const fileUploadInput = debugElement.query(By.css('#fileUploadInput'));
+        expect(fileUploadInput).to.exist;
+        expect(fileUploadInput.nativeElement.disabled).to.be.false;
+
+        // check if fileUploadLabel value is not set
+        const fileUploadLabel = debugElement.query(By.css('.custom-file-label.overflow-ellipsis'));
+        expect(fileUploadLabel).to.exist;
+        expect(fileUploadLabel.nativeElement.value).to.be.undefined;
+
+        // check if extension elements are set
+        const extension = debugElement.query(By.css('.ml-1.badge.badge-info'));
+        expect(extension).to.exist;
+        expect(extension.nativeElement.textContent.replace(/\s/g, '')).to.be.equal(fileUploadExercise.filePattern.split(',')[0].toUpperCase());
+    }));
+
+    it('Submission and file uploaded', fakeAsync(() => {
+        // Ignore window confirm
+        window.confirm = () => {
+            return false;
+        };
+        const fileName = 'exampleSubmission';
+        comp.submissionFile = new File([''], fileName, { type: 'application/pdf' });
+        comp.submission = createFileUploadSubmission();
+        fixture.detectChanges();
+
+        // check if fileUploadLabel value is not set
+        const fileUploadLabel = debugElement.query(By.css('.custom-file-label.overflow-ellipsis'));
+        expect(fileUploadLabel).to.exist;
+        expect(fileUploadLabel.nativeElement.textContent).to.be.equal(fileName);
+
+        let submitFileButton = debugElement.query(By.css('.btn.btn-success'));
+        spyOn(fileUploaderService, 'uploadFile').and.returnValue(Promise.resolve({ path: 'test' }));
+        submitFileButton.nativeElement.click();
+        comp.submission.submitted = true;
+        comp.result = new Result();
+        fixture.detectChanges();
+
+        // check if fileUploadInput is available
+        const fileUploadInput = debugElement.query(By.css('#fileUploadInput'));
+        expect(fileUploadInput).to.not.exist;
+
+        submitFileButton = debugElement.query(By.css('.btn.btn-success'));
+        expect(submitFileButton).to.be.null;
+    }));
+
+    it('Incorrect file type can not be submitted', fakeAsync(() => {
+        // Ignore console errors and window confirm
+        console.error = jest.fn();
+        window.confirm = () => {
```

Comment:
In this case, I don't need to do this since for each test it will load with original functionality and can't be overridden by other test. Also realized that setting `window.confirm` is redundant in this line